### PR TITLE
fix(core):  add redirectURI to the response body of register endpoint /user

### PR DESCRIPTION
**Pull request by simeng-li**
This PR Includes following updates:

1. Add redirectURI to the response body of register endpoint /user
2. Update the ui consent api call using useApi hooks.

@wangsijie @gao-sun

### CODE REVIEW

File: packages/core/src/routes/user.ts [hunk 1]
```diff
@@ -36,13 +37,24 @@ export default function userRoutes<T extends AnonymousRouter>(router: T) {
         passwordEncryptionMethod
       );
 
-      ctx.body = await insertUser({
+      await insertUser({
```
Comment (simeng-li) [inline packages/core/src/routes/user.ts:40]: wondering if any db insert error need to be handled?

Comment (gao-sun) [reply packages/core/src/routes/user.ts:40]: see `packages/core/src/middleware/koa-error-handler.ts`

**Comment by gao-sun** [comment]
overall looks good. one minor issue: the scope position doesn't meeting convention. e.g.:

```
fix: (ui) hide loading status when api throws http error 
```

should be:

```
fix(ui): hide loading status when api throws http error 
```

also the PR title should be:

```
fix(core): add `redirectURI` ...
```

actually capitalization is NOT allowed. this issue is also the reason why the CI for PR title has passed, it thinks the sentence starts with `(` and `Add` is a proper noun.

**Comment by simeng-li** [comment]
Thanks for pointing it out. Reset and fix the commit.

**Review by gao-sun — APPROVED** [review]
lgtm

**Pull request merged**